### PR TITLE
Always use aten.constant_pad_nd for mm padding

### PR DESCRIPTION
**Pull request by oniononion36**
Summary: From experiment, it seems like aten.constant_pad_nd has better QPS compared to torch.cat. The qps gain for ig ctr is ~10%, and ~5% for oc.

Test Plan:
```
buck2 run mode/opt -c fbcode.nvcc_arch=a100 //caffe2/torch/fb/model_transform/experimental/benchmark:mts_gpu_benchmark -- --model-path=manifold://ads_storage_fblearner/tree/user/facebook/fblearner/predictor/585279927/480/gpu_lowering/input.predictor.disagg.gpu.merge --lower-backend=AOT_INDUCTOR
```
```
buck2 run mode/opt //caffe2/torch/fb/model_transform/experimental/benchmark:mts_gpu_benchmark -- --model-path=manifold://ads_storage_fblearner/tree/user/facebook/fblearner/predictor/588102397/1500/gpu_lowering/input.predictor.disagg.gpu.merge --lower-backend=AOT_INDUCTOR
```

Differential Revision: D64271583

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov @rec

### CODE REVIEW

**Comment by oniononion36** [comment]
I think it is due to different lowering, in my gpu traces, I can see that using torch.cat for padding results in a couple more triton_poi_fused kernels and a couple other triton_poi_fused kernels also take way longer to execute.

**Comment by eellison** [comment]
@oniononion36 could we try updating the lowering for aten.cat instead ? see, this function https://github.com/pytorch/pytorch/blob/main/torch/_inductor/lowering.py#L1518

**Comment by oniononion36** [comment]
After offline discussion, we agreed to first proceed with diff while the investigation into aten.cat is still ongoing.

**Review by eellison — APPROVED** [review]
todo: follow up with minimal repro

**Comment by shunting314** [comment]
> The qps gain for ig ctr is ~10%, and ~5% for oc.

Is QPS here e2e QPS or just measuring specific kernels?

**Comment by oniononion36** [comment]
Hi Shunting, the QPS here is e2e. In the min-repro kernel example we used, the QPS gain is ~40%

**Comment by eellison** [comment]
Was talking with @shunting314, my suspicion is that the speedup is due to different benchmarking within pad_mm, bc constant_pad_nd is likely much more optimized for case of large tensor + small fill value compared to torch.cat. In the eager benchmarking, there is an additional aten.zeros kernel as well. 

I imagine the inductor codegen is less affected.
